pkg/sriov/networknodestate_test.go

```diff
@@ -0,0 +1,381 @@
+		testSettings := clients.GetTestClients(append([]runtime.Object{}, networkNodeState))
+		networkNodeStateBuilder := NewNetworkNodeStateBuilder(testSettings, defaultNodeName, defaultNodeNsName)
+		err := networkNodeStateBuilder.Discover()
+		assert.Nil(t, err)
+
+		for _, netInterface := range testCase.netInterface {
+			vfNum, err := networkNodeStateBuilder.GetNumVFs(netInterface.Name)
+			assert.Nil(t, err)
+			assert.Equal(t, netInterface.NumVfs, vfNum)
+		}
+	}
+}
+
+func TestNetworkNodeStateGetDriverName(t *testing.T) {
+	testCases := []struct {
+		netInterface srIovV1.InterfaceExts
+		upInterface  srIovV1.InterfaceExts
+	}{
+		{
+			netInterface: []srIovV1.InterfaceExt{
+				{Name: "eth1", Driver: "Mlx"},
+			},
+		},
+		{
+			netInterface: []srIovV1.InterfaceExt{
+				{Name: "eth1", Driver: "intel"},
+				{Name: "eth2", Driver: "mlx"},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		networkNodeState := buildNodeNetworkStateWithNics(testCase.netInterface)
+		testSettings := clients.GetTestClients(append([]runtime.Object{}, networkNodeState))
+		networkNodeStateBuilder := NewNetworkNodeStateBuilder(testSettings, defaultNodeName, defaultNetNsName)
+		err := networkNodeStateBuilder.Discover()
+		assert.Nil(t, err)
+
+		for _, netInterface := range testCase.netInterface {
+			driver, err := networkNodeStateBuilder.GetDriverName(netInterface.Name)
+			assert.Nil(t, err)
+			assert.Equal(t, netInterface.Driver, driver)
+		}
+	}
+}
+
+func TestNetworkNodeStateGetPciAddress(t *testing.T) {
+	testCases := []struct {
+		netInterface srIovV1.InterfaceExts
+		upInterface  srIovV1.InterfaceExts
+	}{
+		{
+			netInterface: []srIovV1.InterfaceExt{
+				{Name: "eth1", PciAddress: "11111"},
+			},
+		},
+		{
+			netInterface: []srIovV1.InterfaceExt{
+				{Name: "eth1", LinkSpeed: "11111"},
+				{Name: "eth2", PciAddress: "22222"},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		networkNodeState := buildNodeNetworkStateWithNics(testCase.netInterface)
+		testSettings := clients.GetTestClients(append([]runtime.Object{}, networkNodeState))
+		networkNodeStateBuilder := NewNetworkNodeStateBuilder(testSettings, defaultNodeName, defaultNodeNsName)
+		err := networkNodeStateBuilder.Discover()
+		assert.Nil(t, err)
+
+		for _, netInterface := range testCase.netInterface {
+			pciAddress, err := networkNodeStateBuilder.GetPciAddress(netInterface.Name)
+			assert.Nil(t, err)
+			assert.Equal(t, netInterface.PciAddress, pciAddress)
+		}
+	}
+}
+
+func buildNodeNetworkState(name string, nsName string) *srIovV1.SriovNetworkNodeState {
+	return &srIovV1.SriovNetworkNodeState{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: nsName,
+		},
+		Spec:   srIovV1.SriovNetworkNodeStateSpec{},
+		Status: srIovV1.SriovNetworkNodeStateStatus{},
+	}
+}
+
+func buildNodeNetworkStateSyncStatus(name, nsName string, syncStatus string) *srIovV1.SriovNetworkNodeState {
```

Comment:
nit:
in function buildNodeNetworkState above you specify separately the type for name and nsName.
In this function you "save" one time writing the type, but still separate for syncStatus. No consistency.